conans/model/scm.py

```diff
@@ -86,7 +87,10 @@ def excluded_files(self):
     def checkout(self):
         output = ""
         if self._data.type == "git":
-            output += self.repo.clone(url=self._data.url)
+            try:
+                output += self.repo.clone(url=self._data.url, branch=self._data.revision, shallow=True)
```

Comment:
⚠️ This is turning `shallow=True` into the default behavior, right? Can we think about a broken use-case?

Reply:
yeah, that was suggestion to use shallow by default, and fallback to the regular otherwise

Reply:
Oh! Ok, mention this change of the default behavior in the PR text, please. I get surprised by this kind of things if they are not stated before.

Reply:
@jgsogo We decided to take the risk, we don't see it necessarily breaking but we are prepared to introduce opt-in/out in the recipe scm dict maybe.